### ISSUE TITLE
Compilation problem with the button or GPIO interrupt API call,attachInterrupt.  

### ISSUE THREAD

**Issue by mightylastingcode** [platformio/platform-espressif32#822]
I need help.   My button example used to work with the following function.

void detectButtonPushISR(void) {..}
attachInterrupt(digitalPinToInterrupt(BUTTON_PIN), detectButtonPushISR, RISING); 

When I compile now with the new expressif 32 package (3.5.0), I get an error saying this does not exist.  When I try to check the header file for this function, the function api is different.

esp32-hal.gpio.h:

void attachInterrupt(uint8_t pin,  int mode);
void attachInterruptArg(uint8_t pin, void (*)(void*), void * arg, int mode);
void detachInterrupt(uint8_t pin);

So I change it to  attachInterruptArg.  The function API does not exist.   

void detectButtonPushISR(void * param)  { ... }
attachInterruptArg(digitalPinToInterrupt(BUTTON_PIN), detectButtonPushISR, NULL, RISING); 

Please help.   Is it an error in the latest espressif32 package installation with PIO IDE?   Is there any way I could use the previous stable version until I am ready for the latest stable version?  For example,platform = https://github.com/platformio/platform-espressif32.git   (How do I find the git file for previous version?)

Thank you,
Michael

=========================  Compilation Log with attachInterrupt ===========================================
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32dev.html
PLATFORM: Espressif 32 (3.5.0) > Espressif ESP32 Dev Module
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
DEBUG: Current (esp-prog) External (esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES: 
 - framework-arduinoespressif32 @ 3.10006.210326 (1.0.6) 
 - tool-esptoolpy @ 1.30100.210531 (3.1.0) 
 - toolchain-xtensa32 @ 2.50200.97 (5.2.0)
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft

src/main.cpp: In function 'void setup()':
src/main.cpp:45:81: error: invalid conversion from 'void (*)()' to 'int' [-fpermissive]
   attachInterrupt(digitalPinToInterrupt(BUTTON_PIN), detectButtonPushISR, RISING);   // ^^^^|____|^^^^  The rising edge when the button is released.
                                                                                 ^
src/main.cpp:45:81: error: too many arguments to function 'void attachInterrupt(uint8_t, int)'
In file included from /home/michael/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal.h:53:0,
                 from /home/michael/.platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:35,
                 from src/main.cpp:1:
/home/michael/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-gpio.h:81:6: note: declared here
 void attachInterrupt(uint8_t pin,  int mode);
      ^
*** [.pio/build/esp32dev/src/main.cpp.o] Error 1
================================= Compilation Log ==========================================

=========================  Compilation Log with attachInterruptArg ===========================================
/home/michael/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-gpio.c:301:13: error: conflicting types for 'attachInterrupt'
 extern void attachInterrupt(uint8_t pin, voidFuncPtr handler, int mode) __attribute__ ((weak, alias("__attachInterrupt")));
             ^
In file included from /home/michael/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-gpio.c:15:0:
/home/michael/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-gpio.h:81:6: note: previous declaration of 'attachInterrupt' was here
 void attachInterrupt(uint8_t pin,  int mode);
      ^
*** [.pio/build/esp32dev/FrameworkArduino/esp32-hal-gpio.c.o] Error 1
=========================  Compilation Log with attachInterruptArg ===========================================

**Comment by mightylastingcode** [comment]
I found the solution to fix my problem.   First, I uninstalled my current Espressif32 package (version 3.5).  Then, I reinstalled the latest Espressif32 package (version 4.3).   Now, everything worked fine now.  Both attachInterrupt and attachInterruptArg function calls are accepted by the compiler.   

Best
Michael

Here is how I installed the new package.  
![Screenshot from 2022-05-30 11-00-22](https://user-images.githubusercontent.com/20387404/171043158-8c9d6316-81f3-4a31-b276-a9f28274142f.png)
